### PR TITLE
Add rag-opensearch template

### PR DESCRIPTION
**Pull request by rawwar**
Adding rag-opensearch template.

### CODE REVIEW

**Review by dhrubo-os — COMMENTED** [review]
let's fix the check failures.

**Comment by rawwar** [comment]
@efriis , please consider reviewing this PR. Thanks.

File: templates/rag-opensearch/dummy_index_setup.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+client_oai = OpenAI(api_key="<OPENAI_API_KEY")
+
+# Create index
+from opensearchpy import OpenSearch
+
+client = OpenSearch(
+    hosts=[{"host": "localhost", "port": 9200}],
+    http_auth=("admin", "admin"),
```
Comment (dhrubo-os) [inline templates/rag-opensearch/dummy_index_setup.py:34]: Can't we use `OPENSEARCH_USERNAME` & `OPENSEARCH_PASSWORD` here?

File: templates/rag-opensearch/dummy_index_setup.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+]
+
+
+client_oai = OpenAI(api_key="<OPENAI_API_KEY")
+
+# Create index
+from opensearchpy import OpenSearch
+
+client = OpenSearch(
+    hosts=[{"host": "localhost", "port": 9200}],
```
Comment (dhrubo-os) [inline templates/rag-opensearch/dummy_index_setup.py:33]: Can we use `OPENSEARCH_URL` here?

File: templates/rag-opensearch/dummy_index_setup.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# ## Search
+
+# text = "Data scientist"
+# res = client_oai.embeddings.create(input=text, model="text-embedding-ada-002")
+# query = {"size":1,"query":{"knn":{"vector_field":{"vector":res.data[0].embedding,"k":2}}}}
+# # Perform the search
+# response = client.search(index=INDEX_NAME, body=query)
+
+# # Print the response
+# print(response)
```
Comment (dhrubo-os) [inline templates/rag-opensearch/dummy_index_setup.py:85]: Do we need these lines?

**Comment by dhrubo-os** [comment]
@baskaryan, @efriis, Can we please get some attention here to review this PR? Thanks.

**Review by efriis — COMMENTED** [review]
Could you add a pyproject.toml metadata section like the following to populate templates.langchain.com?

https://github.com/langchain-ai/langchain/blob/master/templates/rag-conversation/pyproject.toml#L25-L29

**Pull request merged**